PVTPackage/source/MultiphaseSystem/MultiphaseSystemProperties/FactorMultiphaseSystemProperties.hpp

```diff
@@ -0,0 +1,110 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2020-     GEOSX Contributors
+ * All right reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+#ifndef PVTPACKAGE_FACTORMULTIPHASESYSTEMPROPERTIES_HPP
+#define PVTPACKAGE_FACTORMULTIPHASESYSTEMPROPERTIES_HPP
+
+#include "pvt/pvt.hpp"
+
+#include <map>
+
+namespace PVTPackage
+{
+
+class FactorMultiphaseSystemProperties : public pvt::MultiphaseSystemProperties
```

Comment:
`pvt::MultiphaseSystemProperties` is indeed what we expose.
And in the end, we have one `MultiphaseSystemProperties` for each flash.
But that does not prevent us from using utility intermediate classes for code factorisation.
That's what `FactorMultiphaseSystemProperties` is (note the powerful `Factor` in the name 😉 ).
We could use composition techniques instead of inheritances _e.g._.